drivers/lora/rylrxxx.c

```diff
@@ -0,0 +1,684 @@
+	if (err != 0) {
+		LOG_ERR("error taking operation semaphore: %d", err);
+		return err;
+	}
+
+	/* This is not a user error but the documeted way to cancel async reception in lora api*/
+	if (cb == NULL) {
+		goto bail;
+	}
+
+	if (data->is_tx) {
+		LOG_ERR("radio is configured for tx");
+		err = -EOPNOTSUPP;
+		goto bail;
+	}
+
+	data->async_rx_cb = cb;
+	if (RYLR_IS_ASYNC_OP_PENDING(data->pending_async_flags)) {
+		LOG_ERR("pending async opperation");
+		err = -EBUSY;
+		goto bail;
+	}
+	RYLR_SET_RX_PENDING(data->pending_async_flags);
+
+	return err;
+bail:
+	RYLR_CLEAR_RX_PENDING(data->pending_async_flags);
+	k_sem_give(&data->operation_sem);
+	return err;
+}
+
+int rylr_test_cw(const struct device *dev, uint32_t frequency, int8_t tx_power, uint16_t duration)
+{
+	return -ENOSYS;
+}
+
+static int rylr_init(const struct device *dev)
+{
+	int err = 0;
+	struct rylr_data *data = dev->data;
+	const struct rylr_config *config = dev->config;
+
+	if (!gpio_is_ready_dt(&config->reset)) {
+		return -ENODEV;
+	}
+
+	if (!device_is_ready(config->uart)) {
+		return -ENODEV;
+	}
+
+	err = gpio_pin_configure_dt(&config->reset, config->reset.dt_flags | GPIO_OUTPUT_ACTIVE);
+	if (err != 0) {
+		LOG_ERR("error configuring reset gpio: %d", err);
+		return err;
+	}
+
+	k_msgq_init(&data->rx_msgq, data->msgq_buffer, sizeof(struct rylr_recv_msg),
+		    ARRAY_SIZE(data->msgq_buffer));
+
+	err = k_sem_init(&data->script_sem, 0, 1);
+	if (err != 0) {
+		LOG_ERR("error initializing response semaphore. err=%d", err);
+	}
+
+	err = k_sem_init(&data->operation_sem, 1, 1);
+	if (err != 0) {
+		LOG_ERR("error initializing operation semaphore. err=%d", err);
+	}
+
+	const struct modem_backend_uart_config uart_backend_config = {
+		.uart = config->uart,
+		.receive_buf = data->uart_backend_rx_buff,
+		.receive_buf_size = ARRAY_SIZE(data->uart_backend_rx_buff),
+		.transmit_buf = data->uart_backend_tx_buff,
+		.transmit_buf_size = ARRAY_SIZE(data->uart_backend_tx_buff),
+	};
+
+	data->uart_pipe = modem_backend_uart_init(&data->uart_backend, &uart_backend_config);
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_rx_buf,
+		.receive_buf_size = ARRAY_SIZE(data->chat_rx_buf),
+		.delimiter = "\r\n",
+		.delimiter_size = sizeof("\r\n") - 1,
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+	};
+
+	err = modem_chat_init(&data->chat, &chat_config);
+	if (err != 0) {
+		LOG_ERR("error initializing chat %d", err);
+		return err;
+	}
+
+	err = modem_chat_attach(&data->chat, data->uart_pipe);
+	if (err != 0) {
+		LOG_ERR("error attaching chat %d", err);
+		return err;
+	}
+
+	err = modem_pipe_open(data->uart_pipe);
+	if (err != 0) {
+		LOG_ERR("error opening uart pipe %d", err);
+		return err;
+	}
+
+	err = gpio_pin_set_dt(&config->reset, 1);
```

Comment:
Reset line is already being set to active by the `GPIO_OUTPUT_ACTIVE` flag in `gpio_pin_configure_dt` above.

Reply:
ahh right, I'm actually going to take  GPIO_OUTPUT_ACTIVE out of gpio_pin_configure_dt as it feels saner to me for pin configuration and activation to be two different steps